apm-agent-core/src/main/java/co/elastic/apm/agent/report/HttpUtils.java

```diff
@@ -65,4 +69,34 @@ public static void consumeAndClose(@Nullable HttpURLConnection connection) {
             }
         }
     }
+
+    /**
+     * Rebuilds the provided URL with the default handler, as it may have been overriden by an application server at
+     * runtime and might prevent the agent from properly communicate through HTTP/HTTPS.
+     *
+     * @param url URL to rewrite
+     * @return equivalent URL with the default JDK handler
+     * @throws IllegalArgumentException if protocol is not supported or unable to access default handler
+     */
+    public static URL withDefaultHandler(URL url) {
+        // the default handler for URLs might be overridden by another implementation than the one shipped with the JDK
+        // for example, this happens on Weblogic application server and triggers classloading issues as the agent
+        // is unable to properly use the Weblogic classes as they aren't visible to the agent classloaders.
+
+        String protocol = url.getProtocol();
+        try {
+            return new URL(protocol, url.getHost(), url.getPort(), url.getFile(), handlerForProtocol(protocol));
+        } catch (MalformedURLException e) {
+            throw new IllegalArgumentException(e);
+        }
```

Comment:
My instinct would be to fall back to just returning the original url on any exception. For example, if the `sun.net.www.protocol.%s.Handler` class can't be loaded.
 
```suggestion
        } catch (Exception e) {
            logger.warn("...", e);
            return url;
        }
```